pgjdbc/src/test/java/org/postgresql/test/TestUtil.java

```diff
@@ -980,11 +980,43 @@ public static int getBackendPid(Connection conn) throws SQLException {
     return pid;
   }
 
+  public static boolean isPidAlive(Connection conn, int pid) throws SQLException {
+    String sql = "SELECT EXISTS (SELECT * FROM pg_stat_activity WHERE pid = ?) t";
+    try (PreparedStatement stmt = conn.prepareStatement(sql)) {
+      stmt.setInt(1, pid);
+      ResultSet rs = stmt.executeQuery();
+      rs.next();
+      boolean ret = rs.getBoolean(1);
+      rs.close();
+      return ret;
```

Comment:
try with resources for ResultSet?
```suggestion
      try (ResultSet rs = stmt.executeQuery()) {
        rs.next();
        return rs.getBoolean(1);
      }
```